pandas/core/frame.py

```diff
@@ -10593,23 +10594,30 @@ def corrwith(
                 corrs = {}
                 if numeric_only:
                     cols = self.select_dtypes(include=np.number).columns
-                    ndf = self[cols].values.transpose()
                 else:
                     cols = self.columns
-                    ndf = self.values.transpose()
                 k = other.values
+                k_mask = ~other.isna()
+                if isinstance(k, BaseMaskedArray):
+                    k = k._data
```

Comment:
Why are we special Casing here?

Reply:
Nullable arrays are not supported by rank_1d.

Reply:
You can use the mask too in that case

Reply:
Do you mean we still cast the type, but write rank step as the following?
```python
libalgos.rank_1d(k, mask=nonnull_mask)
```

Reply:
I meant more like k_mask = k._mask